src/python/pants/option/global_options.py

```diff
@@ -1718,6 +1718,19 @@ class GlobalOptions(BootstrapOptions, Subsystem):
         default=[],
     )
 
+    work_dir = StrOption(
```

Comment:
Why does scie-pants need to set an env var?

Reply:
As the current working directory is not preserved. But I just realized that another option might be to not change the working directory, then this wouldn't need to be an option at all.. but then there'd need to be a toggle in `scie-pants` for when to change `cwd` or not.

Reply:
It is. Or should be. The nail gun protocol supports that directly.

Reply:
Oh, I may have drawn hasty conclusions from 

https://github.com/pantsbuild/scie-pants/blob/57e67c8a1933de0fc7d0d07d27ac4a8c3e28f46b/src/main.rs#L290 

seems like it is just for that one process it changes cwd. I'll take another circle at this, thanks for the nudge :)

Reply:
Makes me wonder though what it is that needs `cwd` set for the pants-from-sources mode but not the other..?

Reply:
Not answering that question, but the guts of a pantsd run via ng are:
1. https://github.com/pantsbuild/pants/blob/1a44ab70c2c52a63a7d8210279950c1e6b8eb98b/src/python/pants/bin/remote_pants_runner.py#L165
2. https://github.com/pantsbuild/pants/blob/1a44ab70c2c52a63a7d8210279950c1e6b8eb98b/src/rust/engine/src/externs/nailgun.rs#L34-L65
3. https://github.com/pantsbuild/pants/blob/1a44ab70c2c52a63a7d8210279950c1e6b8eb98b/src/rust/engine/nailgun/src/client.rs#L103-L129

At that point you're in the nails crate, but you can see curdir is gathered from the os and passed in step 3.

The ng protocol is here: http://www.martiansoftware.com/nailgun/protocol.html

Reply:
As to your question:
https://github.com/pantsbuild/scie-pants/blob/57e67c8a1933de0fc7d0d07d27ac4a8c3e28f46b/src/main.rs#L289-L292

Maybe the pants script in the Pants repo doesn't need that? If so, just axe that curdir set.

Reply:
Ah, right - you found that bit. No clue. Time to experiment. Maybe it doesn't or else could be fixed not to need it if it does.

Reply:
Yeah, feel there may be some legacy needs we can cut now.. experimenting hat on ;)

Reply:
Workdir from nailgun is still correct and preserved.

But, I found this
https://github.com/pantsbuild/pants/blob/1a44ab70c2c52a63a7d8210279950c1e6b8eb98b/src/python/pants/pantsd/process_manager.py#L477

So that's good. Means we can pickup the original workdir before changing, as I'd rather keep the `cwd = build root` for now as there may be inherent assumptions about that.

Reply:
https://github.com/pantsbuild/scie-pants/issues/134